### ISSUE TITLE
The most recent aas_getfiles_bystream_multilevel breaks reporting for us.

### ISSUE THREAD

**Issue by jones-michael-s** [rhodricusack/automaticanalysis#120]
Hello all,

Just noticed report generation crashes on us using the latest repo code. I've tracked the problem down to aas_getfiles_bystream_multilevel line 18. The variable _domain_ gets defined as a mixed cell array of strings and integers and Matlab unique (here called from aa_unique) can't handle mixed cell arrays. 

I suspect index{1} is supposed to be a string, but it evaluates to the integer 1 when I run the code.

1) Since we're the only shop running OS X, can somebody verify whether this problem occurs on Linux?

2) If 1 = YES, does anyone want to take a stab at fixing it? This is down in the scheduling guts, and I don't have much knowledge of this part of the code.

Thanks much,
Mike

**Comment by tiborauer** [comment]
Which version do you use?
I have certainly no issue with the recent version.

**Comment by jones-michael-s** [comment]
Just to be sure I'm running the most recent code, I cloned the repo again just now and tried it again. Still broken. 

I should be clear on two points:

1) The analysis runs fine. It's ONLY reporting that's broken for us.

2) The try/catches prevent the code from literally crashing so I shouldn't have said "report generation is crashing on us" -- more accurate would be to say report generation is _failing_ on us. The problem is the variable _domains_ in aas_getfiles_bystream_multilevel. Here's the content with the debugger stopped on line 19:

domains =

  1×4 cell array

    [1]    'session'    'subject'    'study'

That first entry looks suspicious to me. Is this sensible contents for "domain"?

**Comment by tiborauer** [comment]
It seems to me that the actual issue is not in the aas_getfiles_bystream_multilevel but the function calling it. The correct syntax is aas_getfiles_bystream_multilevel(aap, <starting domain>, <index/indices>,  <stream name>, [<source: 'input'|'output' (optional, default: 'input'))>]). Then the first element of the _domains_ would be the <starting domain>.

We should, perhaps, arrange an interactive collaborative debugging (e.g. via Skype and/or shared VNC).

**Comment by jones-michael-s** [comment]
Found the problem! 

I checked the calling functions like you recommended and tracked the problem to provenance.m. We were using an old version (I have to update our local distribution by hand rather than using git, and I missed updating aa_toolbox which is where provenance.m lives). Installing the latest version fixed it. 

aas_getfiles_bystream_multilevel was a red herring because it was a recently-modified file...

Thanks for the help!
